packages/transform/src/template/inlining/companion-file.ts

```diff
@@ -49,7 +49,9 @@ export function calculateCompanionTemplateSpans(
     let backingValue: string | undefined;
     if (targetNode) {
       let moduleName = path.basename(script.filename, path.extname(script.filename));
-      backingValue = `({} as unknown as typeof import('./${moduleName}').default)`;
+      backingValue = useJsDoc
+        ? `(/** @type {typeof import('./${moduleName}').default} */ {})`
```

Comment:
I think we need extra parens here (a quirk of TS's rules for applying JSDoc `@type` declarations to expressions):

```suggestion
        ? `(/** @type {typeof import('./${moduleName}').default} */ ({}))`
```

Reply:
Fixed!